polaris/experimental/_dataset_v2.py

```diff
@@ -132,23 +151,81 @@ def dtypes(self) -> dict[str, np.dtype]:
             dtypes[group] = np.dtype(object)
         return dtypes
 
+    @property
+    def has_zarr_md5sum_manifest(self) -> bool:
+        """Whether the md5sum for this dataset's manifest has been computed and stored."""
+        return self._zarr_md5sum_manifest is not None
+
+    @property
+    def zarr_manifest_path(self) -> str:
+        if not self.has_zarr_md5sum_manifest:
+            manifest_path = self.generate_zarr_manifest()
+            self._zarr_md5sum_manifest_path = manifest_path
+        return self._zarr_md5sum_manifest_path
+
     @computed_field
     @property
-    def zarr_md5sum_manifest(self) -> List[ZarrFileChecksum]:
-        """
-        The Zarr Checksum manifest stores the checksums of all files in a Zarr archive.
-        If the dataset doesn't use Zarr, this will simply return an empty list.
-        """
-        if len(self._zarr_md5sum_manifest) == 0 and not self.has_md5sum:
-            # The manifest is set as an instance variable
-            # as a side-effect of the compute_checksum method
-            self.md5sum = self._compute_checksum()
+    def zarr_md5sum_manifest(self) -> str:
+        """This variable stores the checksum of the Parquet file representing the Zarr manifest"""
+        if not self.has_zarr_md5sum_manifest:
+            manifest_md5 = self._compute_checksum()
+            self._zarr_md5sum_manifest = manifest_md5
+
         return self._zarr_md5sum_manifest
 
     def _compute_checksum(self) -> str:
-        """Compute the checksum of the dataset."""
-        zarr_hash, self._zarr_md5sum_manifest = compute_zarr_checksum(self.zarr_root_path)
-        return zarr_hash.md5
+        """Compute the checksum of the Zarr manifest file."""
+        manifest_md5 = calculate_file_md5(self.zarr_manifest_path)
+        return manifest_md5
+
+    def generate_zarr_manifest(self) -> str:
+        #
+        # Init temporary chunk variables
+        MAX_ROWS_IN_CHUNK = 1000
+        temp_chunks = []
+        temp_chunk_count = 0
+        temp_parquet_paths = []
+
+        # Use temporary directory for holding manifest chunks
+        with TemporaryDirectory() as temp_dir_name:
+            #
+            # Traverse files and directories in Zarr archive
+            for root, dirs, files in os.walk(self.zarr_root_path):
```

Comment:
No, but that won't matter for checking integrity. We're going to just extract out the `path` column and append a `state` column to create the stateful file. The end goal is having all paths with a status of "ready", so any order will do as long as we check all chunks.

Reply:
Right, I see! 

However, it feels counterintuitive that a checksum can change when the content doesn't. It seems that simply sorting `dirs` in place and sorting `files` would do the trick!

What do you think?

Reply:
That's a good point, let's add it. I don't think the performance impact will be noticeable for datasets of the expected/more likely size.

I'm not sure I love this `os.walk` implementation though (as I think about it more). Here's why:
- If there's tons of files at a single directory level (billions), we're going to run out of memory fast because those files names are returned as a list object (`files`). We may need to change to an approach that walks the directory file by file somehow.

Reply:
It seems `os.scandir` would be a nice method to use, but its ordering is not deterministic. Maybe having a deterministic checksum for XXL dataset is not easy to do and should be out of scope. What do you think?

Reply:
I like `os.scandir`, I think that would be great to use here.

And I'm for punting the deterministic checksum until we get a first implementation working. I can't think of any simple, memory efficient ways of guaranteeing the ordering of the files while using an iterator right now. I assume the solution is a bit involved.

I'll make a note to write a ticket so we don't forget to look into this.